test/hotspot/jtreg/runtime/cds/appcds/SignedJar.java

```diff
@@ -42,24 +43,31 @@ public static void main(String[] args) throws Exception {
         // Test class exists in signed JAR
         String signedJar = TestCommon.getTestJar("signed_hello.jar");
         OutputAnalyzer output;
-        output = TestCommon.dump(signedJar, TestCommon.list("Hello"));
-        TestCommon.checkDump(output, "Skipping Hello: Signed JAR");
 
-        // At runtime, the Hello class should be loaded from the jar file
-        // instead of from the shared archive since a class from a signed
-        // jar shouldn't be dumped into the archive.
-        output = TestCommon.exec(signedJar, "-verbose:class", "Hello");
-        String expectedOutput = ".class,load. Hello source: file:.*signed_hello.jar";
+        // "testlambda" is for testing JDK-8276184
+        String[] mainArgs = { "dummy", "testlambda" };
+        String mainClass = "Hello";
+
+        for (String mainArg : mainArgs) {
+            output = TestCommon.dump(signedJar, TestCommon.list(mainClass), mainClass, mainArg);
+            TestCommon.checkDump(output, "Skipping Hello: Signed JAR");
+
+            // At runtime, the Hello class should be loaded from the jar file
+            // instead of from the shared archive since a class from a signed
+            // jar shouldn't be dumped into the archive.
+            output = TestCommon.exec(signedJar, "-verbose:class", mainClass, mainArg);
+            String expectedOutput = ".class,load. Hello source: file:.*signed_hello.jar";
 
-        try {
-            output.shouldMatch(expectedOutput);
-        } catch (Exception e) {
-            TestCommon.checkCommonExecExceptions(output, e);
+            try {
+                output.shouldMatch(expectedOutput);
+            } catch (Exception e) {
+                TestCommon.checkCommonExecExceptions(output, e);
+            }
         }
 
         // Test class exists in both signed JAR and unsigned JAR
         String jars = signedJar + System.getProperty("path.separator") + unsignedJar;
-        output = TestCommon.dump(jars, TestCommon.list("Hello"));
+        output = TestCommon.dump(jars, TestCommon.list(mainClass));
```

Comment:
Not in this case. When the `do_entry()` of `ExcludeDumpTimeSharedClasses` is called, it doesn't know that the lambda proxy is associated with a signed `caller_ik`. If there's a need for a log message, it needs to be added in the `do_entry()` of `CleanupDumpTimeLambdaProxyClassTable`.

Reply:
Checking the exclusion message, even if it's printed, will be more involved, especially if you want to ensure that several such proxy classes are excluded. Since the proxies have similar but non-deterministic names (they have numbers in them), you will need to search for a pattern in the log and count the number of times it appears.

I think it's easier and more reliable to run with `-Xlog:cds+class=debug`, and check that the proxy class is not in the print out.

Reply:
There's a bug in Hello.java which I've fixed in my latest push.
Also updated SignedJar.java to perform more checks on the output.